### PR TITLE
Add FreeAxisF4 target

### PR DESCRIPTION
**Pull request by ot0tot**
Adds target for FreeAxis F4 25A AIO FC based on MATEKF411 target with the addition of a flash chip and BMI270.

Needs to be tested, gyro orientation verified, and FreeAxis vendor/board name approved. 

Solves: https://github.com/betaflight/unified-targets/issues/1149

Discord issue: https://discord.com/channels/868013470023548938/1160296288609390592

Thanks to J. R. @sixtyfive for the factory config https://discord.com/channels/868013470023548938/868013470518505494/1167219861601124392

### CODE REVIEW

**Comment by nerdCopter** [comment]
image

recommended:

windows:
```
# git config --global core.autocrlf true
```
linux/osx:
```
git config --global core.autocrlf input
```

cross-platform:
```
git config --global core.safecrlf warn
git config --global core.whitespace cr-at-eol
git config --global core.filemode false
git config --global help.autocorrect true
```

**Comment by nerdCopter** [comment]
@haslinghuis 
`#define MANUFACTURER_ID   FRXS`
not an official listing yet.  believe they are "Cloners"

**Comment by haslinghuis** [comment]
You were reading my mind :)

See https://github.com/betaflight/unified-targets/pull/1173#discussion_r1374932807

**Comment by ot0tot** [comment]
> image
> 
> recommended:
> 
> windows:
> 
> ```
> # git config --global core.autocrlf true
> ```
> 
> linux/osx:
> 
> ```
> git config --global core.autocrlf input
> ```
> 
> cross-platform:
> 
> ```
> git config --global core.safecrlf warn
> git config --global core.whitespace cr-at-eol
> git config --global core.filemode false
> git config --global help.autocorrect true
> ```

Resolved

File: configs/FREEAXISF4/config.h [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+#define ADC1_DMA_OPT        1
+
+
+#define DEFAULT_DSHOT_BURST DSHOT_DMAR_OFF
+#define DEFAULT_DSHOT_BITBANG DSHOT_BITBANG_OFF
+#define MAG_I2C_INSTANCE (I2CDEV_1)
+#define USE_BARO
+#define BARO_I2C_INSTANCE (I2CDEV_1)
+#define DEFAULT_BARO_DEVICE BARO_NONE
```
Comment (haslinghuis) [inline configs/FREEAXISF4/config.h:91]: Why set to none?
Also we should make the flash default:

```suggestion
#define DEFAULT_BLACKBOX_DEVICE BLACKBOX_DEVICE_FLASH
```

File: configs/FREEAXISF4/config.h [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    TIMER_PIN_MAP( 9, PA8 , 1,  0)
+
+
+#define ADC1_DMA_OPT        1
+
+
+#define DEFAULT_DSHOT_BURST DSHOT_DMAR_OFF
+#define DEFAULT_DSHOT_BITBANG DSHOT_BITBANG_OFF
+#define MAG_I2C_INSTANCE (I2CDEV_1)
+#define USE_BARO
```
Comment (haslinghuis) [inline configs/FREEAXISF4/config.h:89]: Please move to line 34 before `#define USE_BARO_BMP280`

```suggestion
```

**Review by haslinghuis — CHANGES_REQUESTED** [review]

**Review by haslinghuis — DISMISSED** [review]

File: configs/FREEAXISF4/config.h [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+#define FC_TARGET_MCU     STM32F411
+
+#define BOARD_NAME        FREEAXISF4
+#define MANUFACTURER_ID   CLNE
+
+#define USE_ACC
+#define USE_GYRO
+#define USE_ACCGYRO_BMI270
+#define USE_BARO
```
Comment (haslinghuis) [inline configs/FREEAXISF4/config.h:32]: When remove the BARO specfic driver we also should remove this line :)
```suggestion
```

Comment (haslinghuis) [reply configs/FREEAXISF4/config.h:32]: Using this define defies cloud build as we try to isolate code to get maximum space for future development. There has been a lot of effort to configure only the drivers physical present.

**Review by haslinghuis — APPROVED** [review]

**Pull request merged**